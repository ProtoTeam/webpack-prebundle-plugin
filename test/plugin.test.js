const WebpackPrebundlePlugin = require("../lib/index");
const path = require("path");
const fs = require("fs");
const { AsyncSeriesHook } = require("tapable");
const chokidar = require("chokidar");

test("should generate vendor js", (done) => {
  const outputPath = path.resolve(__dirname, "./.output/vendors.js");
  const plugin = new WebpackPrebundlePlugin({
    vendors: {
      entries: [
        path.resolve(__dirname, "./entries/mock.a.js"),
        path.resolve(__dirname, "./entries/mock.b.js"),
      ],
      output: path.resolve(__dirname, "./.output/vendors.js"),
    },
  });

  const mockCompilation = {
    options: {},
  };

  mockCompilation.hooks = {
    beforeRun: new AsyncSeriesHook(["compiler"]),
    watchRun: new AsyncSeriesHook(["compiler"]),
  };

  plugin.apply(mockCompilation);

  mockCompilation.hooks.beforeRun.callAsync({}, () => {
    const content = String(fs.readFileSync(outputPath, { encode: "utf8" }));
    expect(content).toMatchSnapshot();
    done();
  });
});

test("should generate common js", (done) => {
  const outputPath = path.resolve(__dirname, "./.output/commons.js");
  const plugin = new WebpackPrebundlePlugin({
    commons: [
      {
        entry: path.resolve(__dirname, "./entries/mock.a.js"),
        output: outputPath,
      },
    ],
  });

  const mockCompilation = {
    options: {},
  };

  mockCompilation.hooks = {
    beforeRun: new AsyncSeriesHook(["compiler"]),
    watchRun: new AsyncSeriesHook(["compiler"]),
  };

  plugin.apply(mockCompilation);

  mockCompilation.hooks.beforeRun.callAsync({}, () => {
    const content = String(fs.readFileSync(outputPath, { encode: "utf8" }));
    expect(content).toMatchSnapshot();
    done();
  });
});