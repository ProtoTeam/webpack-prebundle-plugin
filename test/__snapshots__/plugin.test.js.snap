// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate common js 1`] = `
"// test/entries/mock.a.js
console.log(\\"from mock a.js\\");
//# sourceMappingURL=commons.js.map
"
`;

exports[`should generate vendor js 1`] = `
"(() => {
  var __create = Object.create;
  var __defProp = Object.defineProperty;
  var __getProtoOf = Object.getPrototypeOf;
  var __hasOwnProp = Object.prototype.hasOwnProperty;
  var __getOwnPropNames = Object.getOwnPropertyNames;
  var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
  var __markAsModule = (target) => __defProp(target, \\"__esModule\\", {value: true});
  var __commonJS = (cb, mod) => () => (mod || cb((mod = {exports: {}}).exports, mod), mod.exports);
  var __reExport = (target, module, desc) => {
    if (module && typeof module === \\"object\\" || typeof module === \\"function\\") {
      for (let key of __getOwnPropNames(module))
        if (!__hasOwnProp.call(target, key) && key !== \\"default\\")
          __defProp(target, key, {get: () => module[key], enumerable: !(desc = __getOwnPropDesc(module, key)) || desc.enumerable});
    }
    return target;
  };
  var __toModule = (module) => {
    return __reExport(__markAsModule(__defProp(module != null ? __create(__getProtoOf(module)) : {}, \\"default\\", module && module.__esModule && \\"default\\" in module ? {get: () => module.default, enumerable: true} : {value: module, enumerable: true})), module);
  };

  // test/entries/mock.a.js
  var require_mock_a = __commonJS(() => {
    console.log(\\"from mock a.js\\");
  });

  // test/entries/mock.b.js
  var require_mock_b = __commonJS(() => {
    console.log(\\"from mock b.js\\");
  });

  // lib/.prebundle/entry.js
  var __WEBPACK_PREBUNDLE_GLOBAL_USERSZINDEXDESKTOPWEBPACKPREBUILDPLUGINTESTENTRIESMOCKAJS = __toModule(require_mock_a());
  var __WEBPACK_PREBUNDLE_GLOBAL_USERSZINDEXDESKTOPWEBPACKPREBUILDPLUGINTESTENTRIESMOCKBJS = __toModule(require_mock_b());
  var wrapper = (m) => {
    if (m.default) {
      m.__esModule = true;
    }
    return m;
  };
  window.__WEBPACK_PREBUNDLE_GLOBAL_USERSZINDEXDESKTOPWEBPACKPREBUILDPLUGINTESTENTRIESMOCKAJS = wrapper(__WEBPACK_PREBUNDLE_GLOBAL_USERSZINDEXDESKTOPWEBPACKPREBUILDPLUGINTESTENTRIESMOCKAJS);
  window.__WEBPACK_PREBUNDLE_GLOBAL_USERSZINDEXDESKTOPWEBPACKPREBUILDPLUGINTESTENTRIESMOCKBJS = wrapper(__WEBPACK_PREBUNDLE_GLOBAL_USERSZINDEXDESKTOPWEBPACKPREBUILDPLUGINTESTENTRIESMOCKBJS);
})();
//# sourceMappingURL=vendors.js.map
"
`;
